core/dtn7/src/routing/erouting/ws_client.rs

```diff
@@ -0,0 +1,112 @@
+use super::Packet;
+use futures::channel::mpsc::{unbounded, UnboundedReceiver, UnboundedSender};
+use futures_util::{future, pin_mut, SinkExt, StreamExt};
+use log::info;
+use serde_json::Result;
+use std::str::FromStr;
+use tokio_tungstenite::connect_async;
+use tokio_tungstenite::tungstenite::Message;
+
+pub enum Command {
+    /// Requests a send of the given packet.
+    SendPacket(Packet),
+    /// Requests the closure of the client.
+    Close,
+}
+
+/// Represents the client session of a external router.
+pub struct Client {
```

Comment:
this client is also only needed for external stuff, right? so it might be moved to an extra crate or into dtn7-plus